### PR TITLE
More tests for float functions and some small tests elsewhere

### PR DESCRIPTION
**Pull request by kshyatt**
A bunch of `floatfuncs.jl` was not tested so I made a new file for it.

`reduce.jl` and `test.jl` had small throws/paths that weren't tested, added tests for those.

Added tests for the environment variable functions in `rounding.jl` which weren't tested fully.

### CODE REVIEW

**Comment by kshyatt** [comment]
Gah! This passed locally. I will fix + update.

File: test/test.jl [hunk 1]
```diff
@@ -90,7 +90,8 @@ end
 @test isapprox(.1+.1+.1, .3)
 @test !isapprox(.1+.1+.1, .4)
 
+@test_throws UndefVarError test_approx_eq(ones(10),ones(11),1e-8,"a","b")
+@test_throws UndefVarError test_approx_eq(ones(10),zeros(10),1e-8,"a","b")
```
Comment (tkelman) [inline test/test.jl]: how are these `UndefVarError`s?

Comment (kshyatt) [inline test/test.jl]: They are throwing a generic `error("stuff")`. I thought the correct thing was `UnderVarError` but maybe it should be `ErrorException` instead?

Comment (tkelman) [inline test/test.jl]: Generic `error` throws an `ErrorException`, this is throwing `UndefVarError` because it isn't qualified as `Test.test_approx_eq`

Comment (kshyatt) [inline test/test.jl]: Oh no!!! I'm sorry. I'll fix it. Thanks for catching this and reviewing my bad PRs :)

**Comment by tkelman** [comment]
lgtm

**Pull request merged**